#! /bin/sh -e

# DP: adjust hrefs to point to the local documentation

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- libstdc++-v3/doc/doxygen/mainpage.html~	2008-02-19 21:08:07.375796572 +0100
+++ libstdc++-v3/doc/doxygen/mainpage.html	2008-02-19 21:14:20.841430216 +0100
@@ -30,8 +30,8 @@
 <p>There are two types of documentation for libstdc++.  One is the
    distribution documentation, which can be read online
    <a href="http://gcc.gnu.org/onlinedocs/libstdc++/index.html">here</a>
-   or offline from the file doc/html/index.html in the library source
-   directory.
+   or offline in the documentation directory
+   <a href="manual/spine.html">here</a>.
 </p>
 
 <p>The other type is the source documentation, of which this is the first page.
@@ -71,9 +71,9 @@
   This style guide can also be viewed on the <a href="http://gcc.gnu.org/onlinedocs/libstdc++/manual/bk01apas04.html">web</a>.
 
 <h2>License, Copyright, and Other Lawyerly Verbosity</h2>
-<p>The libstdc++ documentation is released under
-   <a href="http://gcc.gnu.org/onlinedocs/libstdc++/manual/bk01pt01ch01s02.html">
-   these terms</a>.
+<p>The libstdc++ documentation is released under these terms
+   (<a href="http://gcc.gnu.org/onlinedocs/libstdc++/manual/bk01pt01ch01s02.html">read online</a>, or
+   <a href="manual/bk01pt01ch01s02.html">read offline</a>).
 </p>
 <p>Part of the generated documentation involved comments and notes from
    SGI, who says we gotta say this:
